### ISSUE TITLE
HttpHandler - Cannot set value in unmodifiable Map

### ISSUE THREAD

**Issue by giudon010** [jhomlala/catcher#73]
Hello,
I have some problems when sending report trough http handler.

Following are the stacktrace:
`I/flutter ( 9175): [2020-01-28 09:19:38.822235 | HttpHandler | INFO] HttpHandler error: Unsupported operation: Cannot set value in unmodifiable Map, stackTrace: #0      _ImmutableMap.[]= (dart:core-patch/immutable_map.dart:76:5)
I/flutter ( 9175): #1      _RequestConfig.contentType= (package:dio/src/options.dart:383:12)
I/flutter ( 9175): #2      new _RequestConfig (package:dio/src/options.dart:352:10)
I/flutter ( 9175): #3      new Options (package:dio/src/options.dart:154:8)
I/flutter ( 9175): #4      HttpHandler._sendPost (package:catcher/handlers/http_handler.dart:45:25)
I/flutter ( 9175): #5      HttpHandler.handle (package:catcher/handlers/http_handler.dart:37:14)
I/flutter ( 9175): <asynchronous suspension>
I/flutter ( 9175): #6      Catcher.onActionConfirmed (package:catcher/core/catcher.dart:361:15)
I/flutter ( 9175): #7      ReportMode.onActionConfirmed (package:catcher/model/report_mode.dart:21:23)
I/flutter ( 9175): #8      SilentReportMode.requestAction (package:catcher/mode/silent_report_mode.dart:9:11)
I/flutter ( 9175): #9      Catcher._reportError (package:catcher/core/catcher.dart:321:18)
I/flutter ( 9175): #10     Catcher._setupErrorHooks.<anonymous closure> (package:catcher/core/catcher.dart:149:13)
I/flutter ( 9175): #11     _roo`

I think that there are some problems because declared 'headers' in HttpHandler is final.

Best Regards

**Comment by jhomlala** [comment]
Can you share example code?

**Comment by giudon010** [comment]
Of course.

This is my "main":

```

 void main() async {
 
   CatcherOptions debugOptions = CatcherOptions(
       SilentReportMode(),
       [
         ConsoleHandler(
             enableApplicationParameters: true,
             enableDeviceParameters: true,
             enableCustomParameters: true,
             enableStackTrace: true
         ),
         HttpHandler(
             HttpRequestType.post,
             Uri.parse(StaticUtils.appServerUrl+"/XXX.php"),
             printLogs: true
         ),
       ],
       localizationOptions: [
         LocalizationOptions.buildDefaultItalianOptions(),
       ],
   );
  CatcherOptions releaseOptions = CatcherOptions(
       SilentReportMode(),
       [
         ConsoleHandler(
             enableApplicationParameters: true,
             enableDeviceParameters: true,
             enableCustomParameters: true,
             enableStackTrace: true,
         ),
         HttpHandler(
             HttpRequestType.post,
             Uri.parse(StaticUtils.appServerUrl+"/YYY.php"),
             printLogs: false,
         ),
       ],
       localizationOptions: [
         LocalizationOptions.buildDefaultItalianOptions(),
       ]
   );
 
   Catcher(
     MaterialApp(
     ...
     debugConfig: debugOptions,
     releaseConfig: releaseOptions,
   );
 }

```

I can confirm that I only solved by removing from class HttpHandler the field "headers" from Options:
```

 Future<bool> _sendPost(Report error) async {
     try {
       var json = error.toJson();
       Options options = Options(
           sendTimeout: requestTimeout,
           receiveTimeout: responseTimeout);
       _printLog("Calling: ${endpointUri.toString()}");
       Response response =
           await _dio.post(endpointUri.toString(), data: json, options: options);
       _printLog(
           "HttpHandler response status: ${response.statusCode} body: ${response.data}");
       return SynchronousFuture(true);
     } catch (error, stackTrace) {
       _printLog("HttpHandler error: $error, stackTrace: $stackTrace");
       return SynchronousFuture(false);
     }
   }

```

I think that it can depend from the flutter version.

This is what I get by launching flutter version:

> Flutter 1.12.13+hotfix.5 • channel stable • https://github.com/flutter/flutter.git
> Framework • revision 27321ebbad (3 months ago) • 2019-12-10 18:15:01 -0800
> Engine • revision 2994f7e1e6
> Tools • Dart 2.7.0

**Comment by jhomlala** [comment]
I'm using Flutter (Channel stable, v1.12.13+hotfix.7 and Catcher 0.3.2 and can't reproduce your error.
Can you upgrade your flutter version, upgrade catcher version and do flutter packages upgrade and try again?

**Comment by pantheronleo** [comment]
Also having same issue, after removing headers prop it works.

Flutter 1.15.4-pre.47 • channel master • https://github.com/flutter/flutter.git
Framework • revision b2b97c2a7e (3 weeks ago) • 2020-02-12 17:18:09 -0800
Engine • revision e0ebaea590
Tools • Dart 2.8.0 (build 2.8.0-dev.9.0 e4c39721c4)
Catcher 0.3.2


> Of course.
> 
> This is my "main":
> 
> ```
> 
>  void main() async {
>  
>    CatcherOptions debugOptions = CatcherOptions(
>        SilentReportMode(),
>        [
>          ConsoleHandler(
>              enableApplicationParameters: true,
>              enableDeviceParameters: true,
>              enableCustomParameters: true,
>              enableStackTrace: true
>          ),
>          HttpHandler(
>              HttpRequestType.post,
>              Uri.parse(StaticUtils.appServerUrl+"/XXX.php"),
>              printLogs: true
>          ),
>        ],
>        localizationOptions: [
>          LocalizationOptions.buildDefaultItalianOptions(),
>        ],
>    );
>   CatcherOptions releaseOptions = CatcherOptions(
>        SilentReportMode(),
>        [
>          ConsoleHandler(
>              enableApplicationParameters: true,
>              enableDeviceParameters: true,
>              enableCustomParameters: true,
>              enableStackTrace: true,
>          ),
>          HttpHandler(
>              HttpRequestType.post,
>              Uri.parse(StaticUtils.appServerUrl+"/YYY.php"),
>              printLogs: false,
>          ),
>        ],
>        localizationOptions: [
>          LocalizationOptions.buildDefaultItalianOptions(),
>        ]
>    );
>  
>    Catcher(
>      MaterialApp(
>      ...
>      debugConfig: debugOptions,
>      releaseConfig: releaseOptions,
>    );
>  }
> ```
> 
> I can confirm that I only solved by removing from class HttpHandler the field "headers" from Options:
> 
> ```
> 
>  Future<bool> _sendPost(Report error) async {
>      try {
>        var json = error.toJson();
>        Options options = Options(
>            sendTimeout: requestTimeout,
>            receiveTimeout: responseTimeout);
>        _printLog("Calling: ${endpointUri.toString()}");
>        Response response =
>            await _dio.post(endpointUri.toString(), data: json, options: options);
>        _printLog(
>            "HttpHandler response status: ${response.statusCode} body: ${response.data}");
>        return SynchronousFuture(true);
>      } catch (error, stackTrace) {
>        _printLog("HttpHandler error: $error, stackTrace: $stackTrace");
>        return SynchronousFuture(false);
>      }
>    }
> ```
> 
> I think that it can depend from the flutter version.
> 
> This is what I get by launching flutter version:
> 
> > Flutter 1.12.13+hotfix.5 • channel stable • https://github.com/flutter/flutter.git
> > Framework • revision 27321ebbad (3 months ago) • 2019-12-10 18:15:01 -0800
> > Engine • revision 2994f7e1e6
> > Tools • Dart 2.7.0

**Comment by jhomlala** [comment]
Fixed in Catcher 0.3.4
